### PR TITLE
solution

### PR DESCRIPTION
**Pull request by BilokonYaroslav**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const semyResult = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
```
Comment (OleziO) [inline src/transformStateWithClones.js:15]: Move all cases to the object.

Example of usage:
`case ActionType.AddProperties:`

**Review by OleziO — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js (hunk 1 above)
Comment (BilokonYaroslav) [reply src/transformStateWithClones.js]: I don't really think that i understand correctly, but I hope

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -11,18 +11,18 @@ function transformStateWithClones(state, actions) {
...
         for (const key of action.keysToRemove) {
           delete semyResult[key];
         };
         break;
 
-      case 'clear':
+      case (!action.extraData && !action.keysToRemove):
         for (const key in semyResult) {
           delete semyResult[key];
         };
```
Comment (demetrxx) [inline src/transformStateWithClones.js]:
```
const operations = {
  add: 'addProperties',
  remove: 'removeProperties',
  clear: 'clearProperties',
}

switch (action.type) {
  case(operations.add) {...
  }
  case(operations.remove) ...
}
```

**Review by demetrxx — CHANGES_REQUESTED** [review]
Let's try one more time, keep it up!

File: src/transformStateWithClones.js (hunk 2 above)
Comment (BilokonYaroslav) [reply src/transformStateWithClones.js]: Thank you

**Review by TarasHoliuk — APPROVED** [review]

**Pull request closed without merging**